### PR TITLE
Quote | and && to allow passing it to retry

### PR DESCRIPTION
**Pull request by gaul**

### CODE REVIEW

File: test/integration-test-common.sh [hunk 1]
```diff
@@ -279,11 +279,11 @@ function stop_s3fs {
     # Retry in case file system is in use
     if [ `uname` = "Darwin" ]; then
         if df | grep -q $TEST_BUCKET_MOUNT_POINT_1; then
-            retry 10 df | grep -q $TEST_BUCKET_MOUNT_POINT_1 && umount $TEST_BUCKET_MOUNT_POINT_1
+            retry 10 df | grep -q $TEST_BUCKET_MOUNT_POINT_1 "&&" umount $TEST_BUCKET_MOUNT_POINT_1
```
Comment (gaul) [inline test/integration-test-common.sh:282]: Huh, this fails on macOS but succeeds on Linux.  Some weird issue with operator precedence?

Comment (gaul) [reply test/integration-test-common.sh]: Quoted `|`.

File: test/integration-test-common.sh [hunk 2]
```diff
@@ -279,11 +279,11 @@ function stop_s3fs {
     # Retry in case file system is in use
     if [ `uname` = "Darwin" ]; then
         if df | grep -q $TEST_BUCKET_MOUNT_POINT_1; then
-            retry 10 df | grep -q $TEST_BUCKET_MOUNT_POINT_1 && umount $TEST_BUCKET_MOUNT_POINT_1
+            retry 10 df "|" grep -q $TEST_BUCKET_MOUNT_POINT_1 "&&" umount $TEST_BUCKET_MOUNT_POINT_1
```
Comment (gaul) [inline test/integration-test-common.sh:282]: This is still a problem.  It appears that macOS's older bash parses differently?

Comment (gaul) [reply test/integration-test-common.sh:282]: Ignoring this for now and just changing the Linux path.

**Review by ggtakec — APPROVED** [review]

**Pull request merged**